python/ray/util/metrics.py

```diff
@@ -49,6 +57,20 @@ def __init__(
             if not isinstance(key, str):
                 raise TypeError(f"Tag keys must be str, got {type(key)}.")
 
+    def _is_invalid_metric_name(self, name: str):
```

Comment:
```suggestion
    def _is_invalid_metric_name(self, name: str) -> bool:
```
we really should have type checking enabled but don't :(

Reply:
How come? Everything will break and it's cumbersome to maintain?